notebooks/find-quotes.py

```diff
@@ -0,0 +1,155 @@
+
+
+import marimo
+
+__generated_with = "0.13.2"
+app = marimo.App(width="medium", app_title="Finding General Quotes")
+
+
+@app.cell
+def _():
+    import marimo as mo
+    import polars as pl
+    import re
+    return mo, pl, re
+
+
+@app.cell
+def _(mo):
+    mo.md(
+        """
+        # Finding general quotes in Die Neue Zeit
+        This notebook explores finding quotes within Die Neue Zeit article using simple search rules.
+        Namely, that a quote is text that occurs between the starting quote character „ and an ending quote character (either “ or ").
+        In practice, this identifies quotations as well as titles and concepts.
+
+        **Findings:** This simple approach works when the underlying transcription is accurate and when quotations are not broken across pages. The latter is especially a problem for block quotes.
+        """
+    )
+    return
+
+
+@app.cell
+def _(re):
+    def find_quotes(text):
+        search_text = text
+        results = []
+        # Find basic quotes of the form „ ... “ or „ ... "
+        for match in re.finditer(r"„[^“\"]+[“\"]", search_text):
+            # Add match to results
+            results.append(match)
+            ## White out match
+            start, end = match.span()
+            search_text = (
+                search_text[:start] + " " * (end - start) + search_text[end:]
+            )
+        return results
+    return (find_quotes,)
+
+
+@app.cell
+def _():
+    ## Load quotes and page data
+    return
+
+
+@app.cell
+def _(pl):
+    # Load quotes data
+    quotes_data = pl.read_csv("data/direct_quotes_subset.csv")
+    quotes_data.select(
+        "FILE", "page_index", "TAGS", "QUOTE_TRANSCRIPTION", "start_index"
+    ).sort("page_index", "start_index")
+    return (quotes_data,)
+
+
+@app.cell
+def _(quotes_data):
+    # Convert quote data into page data
+    page_data = (
+        quotes_data.unique("page_text")
+        .select("FILE", "page_index", "page_text")
+        .sort("page_index")
+    )
+    page_data
+    return (page_data,)
+
+
+@app.cell
+def _(mo):
+    mo.md(r"""## View Results by Page""")
+    return
+
+
+@app.cell
+def _(mo, page_data):
+    page_slider = mo.ui.slider(start=1, stop=len(page_data.rows()), value=2)
+    return (page_slider,)
+
+
+@app.cell
+def _(mo, page_slider):
+    mo.vstack(
+        [
+            mo.md("Select Slider:"),
+            page_slider,
+            mo.md(f"Selected Page: {page_slider.value}"),
+        ]
+    )
+    return
```

Comment:
For notebook usability, it's easiest to use when the slider is as close as possible to the content it controls. I moved the slider down one cell and hid code for both the slider and the text output when I was reviewing, which made it easier.